### ISSUE TITLE
uploadOnSave will fail for files on new created folders

### ISSUE THREAD

**Issue by EthraZa** [lukasz-wronski/vscode-ftp-sync#22]
uploadOnSave:true will fail for files on folders that still does not exist in remote server with the error:
~~~
Ftp-sync: Uploading tst.js failed: Error: Can't open that file: No such file or directory
~~~

Using Sync L2R creates the new folder remotely and "fix" everything.

**Comment by lukasz-wronski** [comment]
I was trying on my FTP and cannot reproduce it. There is a mechanism that should check if the remote folder exists and creates it if not (recursively). It works perfectly on my FTP server. Any idea how can I try to reproduce it? Anyone have similar problem?

**Comment by kevinjpetersen** [comment]
I'm getting the same problem.. I tried with a clean local root. Then created a folder, and immediately after that created a file in that folder and entered some text, to be able to save it for "UploadOnSave" mechanism.

I get the same error as above, I guess It doesn't directly checks for every subfolder in the local/remote locations? Or maybe it's something else.

What I guess is the problem, is that nothing is telling the new folder, that it should be a part of the UploadOnSave functionality. Maybe that functionality has been excluded on UploadOnSave, as it DOES create the folder on the remote server, if you use the L2R.

**Comment by lukasz-wronski** [comment]
I just attempt to fix (as I still cannot reproduce it). Can you check if it works now for you?

**Comment by EthraZa** [comment]
No, I still get the error.

**Comment by lukasz-wronski** [comment]
I managed to reproduce this on my other FTP. Fixed now, will be available in `0.2.9` tonight.
